packages/full-stack-tests/src/components/unified-selection/PropertyGrid.test.tsx

```diff
@@ -0,0 +1,130 @@
+
+          // width and height should generally we computed using ResizeObserver API or one of its derivatives
+          const [width] = React.useState(400);
+          const [height] = React.useState(600);
+
+          // data provider is going to be empty is no elements are selected
+          if (numSelectedElements === 0)
+            return <>Select an element to see its properties</>;
+
+          // there's little value in loading properties for many elements (see `PropertyDataProviderWithUnifiedSelectionProps.requestedContentInstancesLimit`)
+          if (isOverLimit)
+            return <>Please select less elements</>;
+
+          // render the property grid
+          return (
+            <VirtualizedPropertyGridWithDataProvider
+              dataProvider={dataProvider}
+              width={width}
+              height={height}
+            />
+          );
+        }
+        // __PUBLISH_EXTRACT_END__
+
+        // set up imodel for the test
+        const elementKeys: InstanceKey[] = [];
+        const imodel = await buildTestIModel(this, (builder) => {
+          const categoryKey = insertSpatialCategory(builder, "My Category");
+          const modelKey = insertPhysicalModel(builder, "My Model");
+          elementKeys.push(
+            insertPhysicalElement(builder, "My Element 1", modelKey.id, categoryKey.id),
+            insertPhysicalElement(builder, "My Element 2", modelKey.id, categoryKey.id),
+          );
+        });
+
+        // render the component
+        const { container, getByText } = render(
+          <MyPropertyGrid imodel={imodel} />
+        );
+        await waitFor(() => getByText("Select an element to see its properties"));
+
+        // test Unified Selection -> Property Grid content synchronization
+        act(() => Presentation.selection.replaceSelection("", imodel, new KeySet([elementKeys[0]])));
+        await ensurePropertyGridHasPropertyRecord(container, "User Label", "My Element 1");
+
+        act(() => Presentation.selection.replaceSelection("", imodel, new KeySet([elementKeys[1]])));
+        await ensurePropertyGridHasPropertyRecord(container, "User Label", "My Element 2");
+      });
+
+    });
+
+  });
+
+});
+
+async function ensurePropertyGridHasPropertyRecord(container: HTMLElement, propertyLabel: string, propertyValue: string) {
+  // find & expand the root category
+  const category = await waitFor(() => getRootPropertyCategory(container));
+  if (!category.querySelector(".iui-expanded"))
+    fireEvent.click(category.querySelector(".iui-expandable-block .iui-header")!);
+  await waitFor(() => expect(category.querySelector(".iui-expanded")).to.not.be.null);
+
+  // find the property label
+  await waitFor(() => {
+    const record = getPropertyRecordByLabel(container, propertyLabel);
+    expect(record.querySelector(`.components-property-record-value [title="${propertyValue}"]`)).to.not.be.null;
```

Comment:
Is it really necessary to check that `.components-property-record-value` with title `propertyValue` is rendered under same parent as `propertyLabel`? Wouldn't it be enought to check that `propertyLabel` and `propertyValue` are rendered with `getByText`?

Reply:
agree, simplified